mysql/datadog_checks/mysql/statements.py

```diff
@@ -184,18 +186,41 @@ def _query_summary_per_statement(self):
 
     def _normalize_queries(self, rows):
         normalized_rows = []
+        event_statements_summary_full = False
+
         for row in rows:
             normalized_row = dict(copy.copy(row))
-            try:
-                obfuscated_statement = datadog_agent.obfuscate_sql(row['digest_text'], self._obfuscate_options)
-            except Exception as e:
-                self.log.warning("Failed to obfuscate query '%s': %s", row['digest_text'], e)
-                continue
 
-            normalized_row['digest_text'] = obfuscated_statement
-            normalized_row['query_signature'] = compute_sql_signature(obfuscated_statement)
+            # A NULL digest indicates that some queries are not captured in metrics because
+            # the performance_schema query limit was hit. It is important to treat this row
+            # as a real query so that % time metrics calculations are still valid for all
+            # queries, not just tracked queries.
+            # See: https://dev.mysql.com/doc/refman/5.7/en/statement-summary-tables.html
+            if normalized_row['digest'] is None:
```

Comment:
This probably works but I'm not sure the edge cases that could could result in null digest. To be fully correct, this should check the schema and digest for null values.